port/nxp/lpc/build.zig

```diff
@@ -1,67 +1,72 @@
+const Chip = internals.Chip;
+const ModuleDeclaration = internals.ModuleDeclaration;
+const regz = @import("microzig/tools/regz");
 
-fn path(comptime suffix: []const u8) std.Build.LazyPath {
-    return .{
-        .cwd_relative = comptime ((std.fs.path.dirname(@src().file) orelse ".") ++ suffix),
-    };
-}
+pub const chips = struct {
+    pub const lpc176x5x = &TargetAlias.init("lpc176x5x");
+};
 
-const hal = .{
-    .root_source_file = path("/src/hals/LPC176x5x.zig"),
+pub const boards = struct {
+    pub const mbed = &TargetAlias.init("mbed");
 };
 
-pub const chips = struct {
-    pub const lpc176x5x = MicroZig.Target{
-        .preferred_format = .elf,
-        .chip = .{
-            // TODO: Separate over those chips, this is not generic!
-            .name = "LPC176x5x",
-            .cpu = MicroZig.cpus.cortex_m3,
-            .memory_regions = &.{
-                .{ .offset = 0x00000000, .length = 512 * 1024, .kind = .flash },
-                .{ .offset = 0x10000000, .length = 32 * 1024, .kind = .ram },
-                .{ .offset = 0x2007C000, .length = 32 * 1024, .kind = .ram },
-            },
-            .register_definition = .{
-                .json = path("/src/chips/LPC176x5x.json"),
-            },
+pub fn build(b: *std.Build) void {
+    const lpc176x5x_chip: Chip = .{
+        .b = b,
+        .name = "LPC176x5x",
+        .cpu = .{
+            .cpu_arch = .thumb,
+            .cpu_model = .{ .explicit = &std.Target.arm.cpu.cortex_m3 },
+            .os_tag = .freestanding,
+            .abi = .eabi,
+        },
+        .register_definition = .{ .json = b.path("src/chips/LPC176x5x.json") },
+        .memory_regions = &.{
+            .{ .offset = 0x00000000, .length = 512 * 1024, .kind = .flash },
+            .{ .offset = 0x10000000, .length = 32 * 1024, .kind = .ram },
+            .{ .offset = 0x2007C000, .length = 32 * 1024, .kind = .ram },
         },
-        .hal = hal,
-        .binary_post_process = postprocess,
     };
-};
 
-pub const boards = struct {
-    pub const mbed = struct {
-        pub const lpc1768 = MicroZig.Target{
-            .preferred_format = .hex,
-            .chip = chips.lpc176x5x.chip,
-            .hal = hal,
-            .board = .{
-                .name = "mbed LPC1768",
-                .url = "https://os.mbed.com/platforms/mbed-LPC1768/",
-                .root_source_file = path("/src/boards/mbed_LPC1768.zig"),
-            },
-            .binary_post_process = postprocess,
-        };
-    };
-};
+    const lpc176x5x_hal = ModuleDeclaration.init(b, .{
+        .root_source_file = b.path("src/hals/LPC176x5x.zig"),
+    });
 
-/// Post-processes an ELF file to add a checksum over the first 8 words so the
+    internals.submit_target(chips.lpc176x5x, .{
+        .chip = lpc176x5x_chip,
+        .hal = lpc176x5x_hal,
+        .preferred_binary_format = .elf,
+        .patch_elf = .{
+            .b = b,
+            .func = patch_elf,
+        },
```

Comment:
Does it make sense that the `patch_elf` holds a reference to the constructing builder?